### ISSUE TITLE
Grad-CAM++ 1d v2.py是否可用？

### ISSUE THREAD

**Issue by Rainxyh** [liguge/1D-Grad-CAM-for-fault-diagnosis#2]
target_layer设为nn.Conv2d(8, 16, 5, padding=2)
在output[target_category]和线性插值处cam = np.interp报错

**Comment by liguge** [comment]
当然是可以用的。这些都经过测试了。具体报什么错误呢？

**Comment by liguge** [comment]
> target_layer设为nn.Conv2d(8, 16, 5, padding=2) 在output[target_category]和线性插值处cam = np.interp报错

还请你debug一下具体的出错的原因，我们这个代码是输入是一位的数据，代码都经过有些人的测试了。

**Comment by Rainxyh** [comment]
您好，请问可以给出一个基本的能够直接运行的测试模型吗？
例如如下网络，以及input_tensor = torch.randn((5, 4096))[0:1, :]作为输入，应该如何进行测试？

class GruNet(nn.Module):

    def __init__(self, input_dim=32, hidden_num=16, output_dim=10, num_layers=3):
        super(GruNet, self).__init__()
        self.input_dim = input_dim
        self.GRU_layer = nn.GRU(batch_first=True, input_size=input_dim, hidden_size=hidden_num, num_layers=num_layers)
        self.linear_1 = nn.Linear(hidden_num, 16)
        self.linear_2 = nn.Linear(16, output_dim)

    def forward(self, _x):
        x, hidden_state = self.GRU_layer(_x)
        x = x[:, -1, :]
        x = self.linear_1(x)
        output = self.linear_2(x)
        return output

**Comment by liguge** [comment]
能给一个CNN的模型嘛？这个只对CNN模型有效，并不能够适用于gru（rnn类）的模型。

在 2022-07-17 16:10:03，Rainxyh ***@***.***> 写道：


您好，请问可以给出一个基本的能够直接运行的测试模型吗？
例如如下网络，以及input_tensor = torch.randn((5, 4096))[0:1, :]作为输入，应该如何进行测试？
class GruNet(nn.Module):
def init(self, input_dim=32, hidden_num=16, output_dim=10, num_layers=3):
super(GruNet, self).init()
self.input_dim = input_dim
self.GRU_layer = nn.GRU(batch_first=True, input_size=input_dim, hidden_size=hidden_num, num_layers=num_layers)
self.linear_1 = nn.Linear(hidden_num, 16)
self.linear_2 = nn.Linear(16, output_dim)

def forward(self, _x):

    x, hidden_state = self.GRU_layer(_x)

    x = x[:, -1, :]

    x = self.linear_1(x)

    output = self.linear_2(x)

    return output

**Comment by Rainxyh** [comment]
请问是只对卷积网络有效吗？可用于Linear吗？最好是您能给出一个标准的模板，一维震动数据不需要经过一些维度变换吗？

class CnnNet(nn.Module):

    def __init__(self):
        super(CnnNet, self).__init__()
        self.conv1 = nn.Conv2d(8, 16, 5, padding=2)
        self.conv2 = nn.Conv2d(16, 32, 3, padding=1)
        self.conv3 = nn.Conv2d(32, 48, 3, padding=1)
        self.conv4 = nn.Conv2d(48, 64, 3, padding=1)
        self.pool = nn.MaxPool2d(2)
        self.fc1 = nn.Linear(4 * 256, 512)
        self.fc2 = nn.Linear(512, 10)

    def forward(self, x):
        x = self.pool(F.relu(self.conv1(x)))
        x = self.pool(F.relu(self.conv2(x)))
        x = self.pool(F.relu(self.conv3(x)))
        x = self.pool(F.relu(self.conv4(x)))
        x = x.view(x.shape[0], -1)
        x = F.relu(self.fc1(x))
        x = self.fc2(x)
        return x

**Comment by liguge** [comment]
加一下我的wx吧，17853131673
讨论一下。。




--
发自我的网易邮箱手机智能版



在 2022-07-17 16:42:27，Rainxyh ***@***.***> 写道：


请问是只对卷积网络有效吗？可用于Linear吗？最好是您能给出一个标准的模板，一维震动数据不需要经过一些维度变换吗？

class CnnNet(nn.Module):

def __init__(self):

    super(CnnNet, self).__init__()

    self.conv1 = nn.Conv2d(8, 16, 5, padding=2)

    self.conv2 = nn.Conv2d(16, 32, 3, padding=1)

    self.conv3 = nn.Conv2d(32, 48, 3, padding=1)

    self.conv4 = nn.Conv2d(48, 64, 3, padding=1)

    self.pool = nn.MaxPool2d(2)

    self.fc1 = nn.Linear(4 * 256, 512)

    self.fc2 = nn.Linear(512, 10)



def forward(self, x):

    x = self.pool(F.relu(self.conv1(x)))

    x = self.pool(F.relu(self.conv2(x)))

    x = self.pool(F.relu(self.conv3(x)))

    x = self.pool(F.relu(self.conv4(x)))

    x = x.view(x.shape[0], -1)

    x = F.relu(self.fc1(x))

    x = self.fc2(x)

    return x
